clconf/template/template_funcs.go

```diff
@@ -281,3 +284,91 @@ func RegexReplace(regex, src, repl string) (string, error) {
 	}
 	return re.ReplaceAllString(src, repl), nil
 }
+
+// sortType accepts an array because the input is an optional name of the type
+// for sorting and the actual implementation methods (getsvs, getksvs) accept
+// varargs so this utility function allows direct call without unpacking.
+func sortType(input []string) (string, error) {
+	r := "string"
+	if len(input) > 0 {
+		r = input[0]
+	}
+	if r != "string" && r != "int" {
+		return "", fmt.Errorf("sort: Type '%s' is not supported (only int, string)", r)
+	}
+	return r, nil
+}
+
+type asInt []string
+
+func (p asInt) Len() int { return len(p) }
+func (p asInt) Less(i, j int) bool {
+	a, aerr := strconv.Atoi(p[i])
+	b, berr := strconv.Atoi(p[j])
+
+	if aerr == nil {
+		if berr == nil {
+			return a < b
+		}
+		return true // Numbers come first
+	} else {
+		if berr == nil {
+			return false
+		}
+	}
+	return p[i] < p[j]
+}
+func (p asInt) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
+
+// sortCopy sorts a copy of the input as specified type (string, int, default: string).
+func sortCopy(v []string, asType ...string) ([]string, error) {
+	sortType, err := sortType(asType)
+	if err != nil {
+		return nil, err
+	}
+
+	sorted := make([]string, len(v))
+	copy(sorted, v)
+	if sortType == "int" {
+		sort.Sort(asInt(sorted))
+	} else {
+		sort.Strings(sorted)
+	}
+	return sorted, nil
+}
+
+func getsvs(s *memkv.Store) func(string, ...string) ([]string, error) {
+	return func(pattern string, asType ...string) ([]string, error) {
+		vals, err := s.GetAllValues(pattern)
+		if err != nil {
+			return nil, err
+		}
+		return sortCopy(vals, asType...)
```

Comment:
The same could be said here, why make a copy?  The argument against is that there is no _contractual_ guarantee from `memkv` that `GetAllValues` does not simply return a pointer to an internal structural array.  A simple check of the source [shows it returns a new array](https://github.com/kelseyhightower/memkv/blob/master/store.go#L116-L128) (already sorted).  I think we would be safe in arguing that _returning an internal rep would be a bug_ and that there is no _logical_ reason they would return a mutable reference to their own internal representation, but if we want to make a copy here, its at least _minimally justified_ (though i probably wouldn't for principle).

Reply:
My thought process is that since this must return a value (in order to be consumed by the template) the implication is you're returning something new. Typically when you modify something in-place you don't return the thing you manipulated in place. It also helped out when I was sorting arbitrary interfaces originally, since I had to make a `[]string` anyway.

That said, I'm good either way. I can switch to in-place if we want.

Reply:
[Switched to in-place sorting](https://github.com/pastdev/clconf/pull/26/commits/49ad4b37dfdca4f19fb9029c6dbace6628e2fd79)